src/DataFramesMeta.jl

```diff
@@ -583,6 +584,59 @@ macro transform(x, args...)
     esc(transform_helper(x, args...))
 end
 
+##############################################################################
+##
+## transform! & @transform!
+##
+##############################################################################
+
+
+function transform!_helper(x, args...)
+
+    t = (fun_to_vec(arg) for arg in args)
+
+    quote
+        $DataFrames.transform!($x, $(t...))
+    end
+end
+
+"""
+    @transform!(d, i...)
+
+Mutate `d` inplace to add additional columns or keys based on keyword arguments and return it. 
+
+### Arguments
+
+* `d` : an `AbstractDataFrame`, or `GroupedDataFrame`
+* `i...` : keyword arguments defining new columns or keys
+
+### Returns
+
+* `::AbstractDataFrame` or `::GroupedDataFrame`
+
+### Examples
+
+```jldoctest
+julia> using DataFramesMeta
+
+julia> df = DataFrame(A = 1:3, B = [2, 1, 2]);
+
+julia> @transform!(df, a = 2 * :A, x = :A .+ :B);
+
```

Comment:
Can you add an example in the doc string showing that the columns aren't copied? i.e. 

```
newdf = @transform!(df, x = :A)
newdf.x === df.a
```